tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/AspNetCoreTestFixture.cs

```diff
@@ -231,13 +231,13 @@ private async Task EnsureServerStarted(bool sendHealthCheck)
 
                         if (responseCode == HttpStatusCode.OK)
                         {
-                            Agent.WaitForSpans(1, minDateTime: dateTime);
+                            await Agent.WaitForSpansAsync(1, minDateTime: dateTime).ConfigureAwait(false);
```

Comment:
I guess we _can_ use `ConfigureAwait` here because the method itself will be awaited, and get us back on the right sync context, but I also wonder if we _should_ 🤔 Mostly I'm worried about people copy-pasting it and using it somewhere they shouldn't

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/7145/commits/140b6ac9b377428380005da150f82e712b168cc1